float32 x_left
float32 y_left
float32 x_right
float32 y_right
bool up
bool down
bool left
bool right
bool triangle
bool x
bool square
bool circle
bool select
bool start
bool playstation